sound/soc/sof/sof-audio.c

```diff
@@ -257,6 +257,7 @@ static int sof_setup_pipeline_connections(struct snd_sof_dev *sdev,
 					  struct snd_soc_dapm_widget_list *list, int dir)
 {
 	struct snd_soc_dapm_widget *widget;
+	struct snd_sof_route *sroute;
```

Comment:
The commit message is oddly wrapped at around 50 characters?

Reply:
also, the fixes tag should be not part of the patch commit message.

Reply:
to be clearer, a Fixes tag should refer to an upstream commit with the SHA ("commit title") format.

What you want to use here is 

````
Link: https://github.com/thesofproject/linux/issues/4042
````

Reply:
commit message: the term "feedback route" is very vague for the reviewer. You are using it with a very specific example in mind, but you do not provide any context.

During route setup, playback stream routes are setup by walking through the graph following the sink paths.
Capture routes are conversely set-up by following the source paths (walking backwards from sink to source).

When the topology graph includes a loopback from capture to playback, such as the amplifier feedback provided to a speaker protection, the loopback route is not set-up using the regular graph walks.

This patch adds support ... explain the rest, I couldn't figure out what you were trying to say.
by going through sink path, capture stream routes
are set up by going through source path